src/canonical/handler.js

```diff
@@ -352,7 +352,7 @@ export function validateCanonicalFormat(canonicalUrl, baseUrl, log) {
     }
 
     // Check if the canonical URL has the same domain as the base URL
-    if (url.hostname !== base.hostname) {
+    if (url.hostname !== base.hostname && url.hostname !== `www.${base.hostname}` && `www.${url.hostname}` !== base.hostname) {
```

Comment:
i think some of the functions in `spacecat-shared-utils`, like the compose and strip functions may apply here instead.

Reply:
I've used composeBaseURL